### PR TITLE
CBG-701: Added context and changed logging

### PR DESCRIPTION
**Pull request by JRascagneres**
- Add HTTP context ID to sgcollect logging
- Fixed erroneous error return in sgcollect.go.
- Renamed global `err` global error introduced in sgcollect.go.

### CODE REVIEW

File: rest/sgcollect.go [hunk 1]
```diff
@@ -57,7 +61,7 @@ func (sg *sgCollect) Start(zipFilename string, params sgCollectOptions) error {
 
 	// Return error if there is any failure while obtaining sgCollectPaths.
 	if sg.pathError != nil {
-		return err
+		return sg.pathError
```
Comment (bbrks) [inline rest/sgcollect.go:62]: nice catch

File: rest/sgcollect.go [hunk 2]
```diff
@@ -82,9 +86,10 @@ func (sg *sgCollect) Start(zipFilename string, params sgCollectOptions) error {
 	args = append(args, "--sync-gateway-executable", sgPath)
 	args = append(args, zipPath)
 
-	ctx, cancelFunc := context.WithCancel(context.Background())
-	sg.cancel = cancelFunc
-	cmd := exec.CommandContext(ctx, sgCollectPath, args...)
+	sg.context = context.WithValue(context.Background(), base.LogContextKey{}, base.LogContext{CorrelationID: fmt.Sprintf("SGCollect-%s", strconv.Itoa(rand.Intn(65536)))})
+
+	sg.context, sg.cancel = context.WithCancel(sg.context)
```
Comment (bbrks) [inline rest/sgcollect.go]: A couple of nitpicks, but also want to test the new multiline suggestions:

```suggestion
	ctx := context.WithValue(context.Background(), base.LogContextKey{}, base.LogContext{CorrelationID: "SGCollect-" + base.GenerateRandomID()})

	sg.context, sg.cancel = context.WithCancel(ctx)
```

**Review by bbrks — CHANGES_REQUESTED** [review]
Pretty much looks good to go. Left one suggestion.

File: rest/sgcollect.go (hunk 2 above)
Comment (JRascagneres) [reply rest/sgcollect.go]: Haha at the multiline suggestions...
Do we actually want that RandomID? Seems unnecessarily long?
```
2020-03-18T16:29:41.963Z [INF] c:SGCollect-25ddbd1bf5d4ea655e3b0b4f1e6127c5 sgcollect_info: ntp time (ntpdate -q pool.ntp.org || nc time.nist.gov 13 || netcat time.nist.gov 13) - OK
```
Guess it doesn't really matter all that much though.

**Review by bbrks — APPROVED** [review]
LGTM

**Pull request merged**